## Description
The first step to create a workflow for assembling Golden mussels' genomes is of course to choose the softwares that will be used in the pipeline. 

### Assembly
This step is the ultimate goal of the pipeline. Since we already have assembled the Golden mussel's genome, we can use it as a reference to assembly the next genomes. Our task is to choose three reference-guided assemblers to test on the data that will be produced. Just as a reminder, we will choose different infestation locations in Brazil and sequence three golden mussel's from each location. Each of these three golden mussels will have their genomes assembled using a different assembler, so that we can compare the different strategies and find out which one is the most suitable for future assemblies. 
<pre>
golden mussel 1  --------- assembler 1 ---------> genome  {  Place   
golden mussel 2  --------- assembler 2 ---------> genome  {    -  
golden mussel 3  --------- assembler 3 ---------> genome  {    1      
   
golden mussel 1  --------- assembler 1 ---------> genome  {  Place 
golden mussel 2  --------- assembler 2 ---------> genome  {    -
golden mussel 3  --------- assembler 3 ---------> genome  {    2

golden mussel 1  --------- assembler 1 ---------> genome  {  Place 
golden mussel 2  --------- assembler 2 ---------> genome  {    -
golden mussel 3  --------- assembler 3 ---------> genome  {    3
</pre>

#### Reference papers:  
* [This](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4629036/) paper was used as the main reference to search for reference-guided assemblers: 
 
    > On the other hand, various methods have been proposed for reference-assisted genome assembly, such as Amos [9], RACA [10], ARACHNE [11, 12], IMR/DENOM [13], RAGOUT [14], AlignGraph [15], and the pipeline developed by Gnerre et al. [16] which was developed inside the framework provided by ARACHNE.
    
* Other interesting papers:  
    - [this paper](https://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-15-180)  
    - [Hlishcer paper](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-017-1911-6) to see more options  

#### Most promosing assemblers:
* [IMR](http://mtweb.cs.ucl.ac.uk/mus/www/19genomes/IMR-DENOM/description.html):  
    IMR is the programme that makes the reference-guided assembly, while DENOM takes de novo contigs and uses the reference to make them more contiguous. The MCMERGE programme integrates variant calls from the mapping-based (IMR) and Denovo-assembly-based (DENOM) methods. Thus, as we are only looking for a reference-guided strategy, IMR is the programme that we should choose:

    > "IMR is a programme that iteratively assembles the short reads generated by Illumina sequencers. The novel aspect of IMR is how, starting from the reference sequence, it iteratively mutates it towards the sequence of the sample, and the algorithms used for variant calling."
  
    Potential issue: IMR was designed to reassemble homozygous genomes, and the golden mussel's genome is diployd and with high heterogosity (as shown by the double-peak of k-mers [here](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5836269/figure/fig1/))  

    Pros: the software is downloadable as precompiled Linux binaries, it has already been used to assembly [Arabidopsis thaliana](https://www.nature.com/articles/nature10414) and [mouse](https://www.nature.com/articles/nature10413)'s genomes  

* [AMOScmp](https://www.cse.unr.edu/~bioinfor/papers/genome-assembly-newpaper.pdf):  
    In the AMOS Comparative Assembler (AMOS-Cmp), the reads are aligned to the reference genome using a modified version of the MUMmer algorithm. Although the original paper was tested on small datasets of bacterial genomes,it opens the door to eukaryotic genomes: 
  
    > "The assembly process currently takes more than one week for large eukaryotes and, frequently, multiple assemblies are performed in order to produce the best possible reconstruction of the genome. Comparative assembly will dramatically speed up this process when multiple strains of the same large eukaryote are being sequenced."    

    It was the first comparative assembler:  
    > "Since its launch in 2004, the open-source AMOS project has released several innovative DNA sequence analysis applications including: Hawkeye, a visual analytics tool for inspecting the structure of genome assemblies; the Assembly Forensics and FRCurve pipelines for systematically evaluating the quality of a genome assembly; and AMOScmp, the first comparative genome assembler." ([Schatz, 2013](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3603210/))

    It also seems to deal well with differences between haplotypes, although it does not explicitly mention anything regarding heterozygozity: 

    > "The comparative assembly method is specifically designed to handle polymorphisms between DNA molecules, therefore it can gracefully handle the presence of multiple, divergent, haplotypes in the data being assembled."

    The output from AMOScmp will be a .TIGR () and a .FASTA (that contains all contigs produced by AMOScmp) file. (read the AMOScmp.README file at the doc/ folder, inside amos-3.1.0's distribution folder)  

    Papers using AMOScmp:  
    > "In the second module, multiple genomes are selected as references and short reads are assembled using the comparative assembler AMOScmp" ([Yan, 2011](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3268122/))  

    > "We used a variety of assembly strategies, including the Celera Assembler [8] and AMOScmp [9]. We obtained the best result using AMOScmp on a modified version of the reference sequence, CP000437." ([Puiu, 2008](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2561293/))  

    > "Comparative sequence assembly was performed by AMOScmp program (Pop et al., 2004) as a conservative method that uses the most similar available complete genome sequence as a reference to assemble the 454 reads." ([Sheibani-Tezerji, 2015](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4428218/))  

    Pros: it seems easy to install and run.   
    Cons: I didn't find any system requirements information neither in its website nor in its documentation. Also, it looks like it doesn't support multithread, what can make our assembly processo much slower:  

    > "For example, when using 300 thousand reads, producing the edited reference took approximately 1 hour for our approach, whereas it took 16 hours for Amos."   
"(...) much of the observed difference may be attributed to the fact that Velvet, which is used by default as assembler in the proposed pipeline, is able to run in parallel on multiple cores, whereas Amos can be used on one core at a time." ([Buza, 2015](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4629036/)) 

* [ARACHNE](ftp://ftp.broadinstitute.org/pub/papers/comprd/assisted_assembly/Arachne/doc/Manual.html)  
    The problem with ARACHNE is that it assumes we have data from a single haplotype:

    > "We explain here the assumptions Arachne makes about your sequence reads.   
    These reads come from the entire genome of an organism or a cloned fragment thereof (but not both simultaneously), which we call the target. The target should be from a single haplotype: Arachne does not support the assembly of polymorphic data at this time."  

    Pros: its [paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC155255/) reports system requirements and runtime for a Drosophila genome: "The assembly was rapid: the Drosophila assembly required only 21 hours on a single 667 MHz processor and used 8.4 Gb of memory."

* [MIRA](http://mira-assembler.sourceforge.net/docs/DefinitiveGuideToMIRA.html#chap_dataprep)  
    - It is already installed on Galaxy and it has a reference-guided option, but as they advice: "As a general rule of thumb: if you have an organism with more than 100 to 150 megabases or more than 20 to 40 million reads, you might want to try other assemblers first."  
    - Papers that used MIRA's reference-guided assembly:  

        > "Reference-guided assembly was then used to reconstruct the plastid genomes with the programs MIRA 4.0.2 (Chevreux et al., 2004) and MITObim v1.7 (Hahn et al., 2013). In the process, plastid genomes of Q. rubra (JX970937), Q. aliena (KU240007), and C. mollissima (HQ336406) were used as reference genomes." ([Yang, 2018](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5800003/))  
    - Already started a workflow on Galaxy, called "Testing_MIRA_worflow"
    - Interesting: MIRA has a tool (miramem) to estimate memory requirements. A previous estimate:  
```console   
    Is it a genome or transcript (EST/tag/etc.) project? (g/e/) [g] g
    Size of genome? [4.5m] 1.6g
    Looks like a larger eukaryote, guessing largest chromosome size: 30m
    Change if needed!
    Size of largest chromosome? [30000000] 50000000
    Is it a denovo or mapping assembly? (d/m/) [d] m
    Number of Sanger reads? [0] 0
    Are there 454 reads? (y/n/) [n] n
    Are there PacBio reads? (y/n/) [n] n
    Are there Solexa reads? (y/n/) [n] y
    Number of Solexa reads? [0] 40m
    Average Solexe length? [75] 200

    ************************* Estimates *************************

    The contigs will have an average coverage of ~ 4.4 (+/- 10%)
    Estimates may be way off for pathological cases.

    RAM estimates:
           reads+contigs (unavoidable): 308.8 GiB
                large tables (tunable): 1.2 GiB
                                        ---------
                          total (peak): 310.0 GiB

            add if using -CL:pvlc=yes : 61.5 GiB
            add if setting -CO:msr=no : 123.5 GiB

    Note that some algorithms might try to grab more memory if the need arises and the system has enough RAM. 
    The options for automatic memory management control this:
    -AS:amm, -AS:kpmf, -AS:mps
    Further switches that might reduce RAM (at cost of run time or accuracy):
    -SK:mhim, -SK:mchr (both runtime); -SK:mhpr (accuracy)
\*************************************************************\
```  

#### Assemblers that aren't exactly what we are looking for:  
* [RACA](https://github.com/ma-compbio/RACA): it doesn't seem to do a regular comparative assembly: "Herein, we describe a method, called reference-assisted chromosome assembly (RACA), which can be used to further assemble de novo assembled sequence scaffolds into longer chromosomal fragments."     
* [RAGOUT](https://www.ncbi.nlm.nih.gov/pubmed/24931998): it doesn't seem to be suitable for eukaryotic genomes: "Ragout-a reference-assisted assembly tool for bacterial genomes."   
* [AlignGraph](https://www.ncbi.nlm.nih.gov/pubmed/24932000): it doesn't seem to do a regular comparative assembly: "Here we introduce AlignGraph, an algorithm for extending and joining de novo-assembled contigs or scaffolds guided by closely related reference genomes"  
* [Gnerre et. al](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2745769/): it doesn't seem to do a regular comparative assembly: "Assisted assembly: how to improve a de novo genome assembly by using related species"  
* [e-RGA](http://journal.embnet.org/index.php/embnetjournal/article/view/208/484)  
    The enhanced Reference Guided Assembly (e-RGA) was developed to deal with complex, high heterozygotic genomes:   

    > "(...) in this work, we show how the same pipeline can be effectively used for the assembly of large, highly repetitive and heterozygous genomes (plant genomes)."  

    It merges de novo and reference-guided assemblies to assemble complex genomes. Then, it is not a pure RGA. Another weaknness is that the datasets described in the paper had very high coverage (85 and 90X) and the genomes were relatively small compared to the golden mussel: up to 480Mbp.

#### What if Galaxy doesn't have one of the chosen assemblers?
In that case, we will have to add the tool to Galaxy ourselves. There's a quick [tutorial](https://galaxyproject.org/admin/tools/add-tool-tutorial/) on how to do it, and it doesn't look like it's something difficult to accomplish. 

### To do  
Find examples of comparative assemblies for eukaryotic, large genomes with high heterozygosity
