#### 08-03-2019  
##### RAD-seq for genetic diversity analysis
Once we have a better reference genome, it will be used for mapping variants from different golden mussels specimens.  
A sequencing strategy that seems particularly interesting for that genetic diversity analysis is the restriction site-associated DNA sequencing (*RAD-seq*), which simplifies the original genomes and allows a cost-effective sequencing of many different specimens (sometimes in the order of hundreds of samples).  

##### Finding a good test dataset  
Before developing a workflow for running genetic diversity analysis with RAD-seq data, we needed to find a dataset that could be used to test the workflow.  
Ideally, this dataset had to be published so we can compare the results and also have a SRA page so we could download the raw data. The paper ["Reduced-representation sequencing identifies small effective population sizes of Anopheles gambiae in the north-western Lake Victoria basin, Uganda"](https://malariajournal.biomedcentral.com/articles/10.1186/s12936-018-2432-0#MOESM2) follow those two rules and the SRA page can be found [here](https://www.ncbi.nlm.nih.gov/sra/SRX5121665[accn]).  
Interestingly, the paper has a [summary description](https://static-content.springer.com/esm/art%3A10.1186%2Fs12936-018-2432-0/MediaObjects/12936_2018_2432_MOESM2_ESM.pdf) of all the bioinformatics steps done for the diversity analysis.



