#### 2019-03-08  
##### Creating a Galaxy workflow for RADseq analysis  
Because our GCloud resources were blocked, we started building the workflow for analysing the RADseq data at the [Galaxy public server](https://usegalaxy.org/). The workflow was named RADseq-analysis, but it is not complete yet (it doesn't cover all steps as done in the paper of the original work).  

#### 2019-03-22  
##### Modifying the workflow for RADseq analysis  
Since Galaxy no longer supports the GATK Unified Genotyper and GATK Variant Filtration tools. Instead, we will follow the [pipeline](https://galaxyproject.github.io/training-material/topics/variant-analysis/tutorials/dip/tutorial.html) for variant calling analysis recommended by the Galaxy project. We startet testing the Galaxy-recommended pipeline (created as a workflow named *SNP calling on paired end data - v3*), and can manage to execute it successfully until the FreeBayes step, i.e., we can already build a *VCF* file. Other steps are to test the software SnpEff to annotate our variant calling file and then the software GEMINI, to make queries to the annotated *VCF*.  

#### To Do  
- Import genbank file from l-fortunei whole genome as a SnpEff database. 
- Install and execute GEMINI
