#### 02-11-2019  
Despite being a useful resource for initial biological prospection, our first version of the Limnoperna fortunei genome (v1, assembled by Marcela) is still highly fragmented, what doesn't make it a good template for future assemblies. Then, we've decided to invest effort into making an improved assembly- a genome v2.0. In order to chosse the best strategy for making such improvement, we've read the most recent papers that had the same goal:  

|Paper|Original Genome Strategy|Original Genome Metrics|Strategy for Improving Original Genome|Improved Genome Metrics|Highlights|
|---|---|---|---|---|---|
|[Mu et al.](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6240063/)|Illumina HiSeq 2000|**N50**: 1.03 Mb, **%reads mapped to assembly**: 95,6%|PacBio, ~25X coverage|**N50**: 6.55 Mb, **%reads mapped to assembly**: 99,9%||
|[Kalbfleisch et al.](https://www.ncbi.nlm.nih.gov/pubmed/30456315)|Sanger sequencing, bacterial artificial chromosome (BAC) end pairs, radiation hybrid mapping, and fluorescence in situ hybridization (FISH) mapping|**N50**: 46 Mb|Illumina short reads, Chicago and Hi-C proximity ligation libraries, PacBio long reads, and 10× Chromium linked reads.|**N50**: 86 Mb||
|[Xia et al.](https://www.ncbi.nlm.nih.gov/pubmed/29893829)|Illumina platforms 36.9x coverage|**N50**: ~ 36Kb|PacBio, ~∼73.8x coverage, Illumina, ~50x coverage|**N50**: 432 Kb||
|[Ruggieri et al.](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5967340/)|454, Sanger, BAC |**N50**: 4.68 Mbp, **scaffolds**: 1594|optical mapping approach|**N50**: ,**N50**: 15.52 Mb, **scaffolds**: 143| | 
|[Holt et al.](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5940132/)|Illumina HiSeq2000|**N50**: 3,15 Mb|“Chicago” method -> Illumina 290x coverage|**N50**: ,**N50**: 14 Mb, **scaffolds**: 143||  
|[Zimin et al.](https://www.ncbi.nlm.nih.gov/pubmed/29020755)|Illumina short reads|**contigs**: > 11 millions, **N50**: 8,2 Kb|PacBio + Illumina|**N50**: 107 Kb||  
|[Pasini et al.](https://www.ncbi.nlm.nih.gov/pubmed/28748222)|454 and Illumina at ~160x coverage|**scaffolds**: 14, **small unassigned contigs**: 1649|PacBio|**pieces**(scaffolds + contigs): 56, **new genes identified**: > 1000||
|[Feulner et al.](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6027883/)|Illumina|7236 unordered scaffolds|anchoring to linkage genetic map|383 scaffolds were anchored to the linkage map (representing almost 80% of the total sequence content of the original assembly), the remaining 6,853 could not be anchored||  
|[Argyris et al.](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4316794/)|Illumina, PacBio, Linkage Map|**N50**: 4.68 Mb|Deeper Linkage Map|Mis-assemblies of scaffolds corrected (chimeric), 99/142 scaffolds were anchored (90.5% of total sequence content from assembly)|"The assembly of genomic scaffolds to generate chromosome-scale sequences is done by integrating them with genetic or physical maps [[13]](https://www.ncbi.nlm.nih.gov/pubmed/19596977/)"| 
|[Rice et al.](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5411764/)|Illumina reads, assisted with some BACs|**N50**: 508 kbp|Chicago library preparation to a Illumina HiSeq2500|**N50**: > 13 Mbp||
|[Tørresen et al.](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5241972/); *PS*: genome size ~613 Mbp|454 pyrosequencing ~40x + 0.1x Sanger BAC-ends|**N50 scaffolds**: 0.14 kbp|old 454+Sanger BAC-ends, Illumina ~480x and PacBio ~19x|**N50 scaffolds**: 1.15 Mbp|"A more contiguous reference genome for Atlantic cod will facilitate re-sequencing efforts addressing population genomics investigations, including the detection of structural variants, introgression and hybridization between species(...). Moreover, it will also enable an annotation with more complete gene models and allow for a better understanding of the lack of sequence contiguity in gadMor1"|  
|[Li et al.](https://www.ncbi.nlm.nih.gov/pubmed/27503335), *PS*: genome size ~702Mbp|454 ~16x + Sanger|**N50**: 16.7 kb|Illumina ~102x, PacBio ~29x|**N50**: 111 kb|"(...) Although a useful resource, this assembly covers only ~89 % of the non-repetitive portion of the genome, and has a relatively short (16.7 kb) contig N50 length. These downsides make it difficult to apply this reference in transcriptive or whole-genome re-sequencing analyses."|  
|[Krishnan et al.](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4938638/)|454|---|Illumina + PacBio|---|---|  
|[Warren et al.](https://onlinelibrary.wiley.com/doi/full/10.1111/tpj.12886)|---|**scaffold NG50**: 41.9 kbp; **scaffold N50**: 20.4 kbp|"we re‐scaffolded the previously published V2 PG29 draft genome sequence using a reference PG29 RNA‐seq transcriptome assembly, a set of 27 720 white spruce cDNA clone sequences (Rigault et al., 2011), and large‐fragment (3, 8, 12 kbp) mate pair sequences"|**scaffold NG50**: 71.5 kbp; **scaffold N50**: 34.4 kb|---|
